infra/src/dsl/dsl.ts

```diff
@@ -195,10 +199,13 @@ export class ServiceBuilder<ServiceType> {
    * To provision secrets in the Parameter Store, you need to get in touch with the DevOps team.
    * @param secrets Maps of secret names and their corresponding paths
    */
-  secrets(secrets: Secrets) {
+  secrets<Secs extends Secrets>(secrets: Secs) {
     this.assertUnset(this.serviceDef.secrets, secrets)
     this.serviceDef.secrets = { ...this.serviceDef.secrets, ...secrets }
-    return this
+    return (this as unknown) as ServiceBuilder<
```

Comment:
We had some issues when migrating to node 18 and the upgraded TS version when functions return `unknown` since TS and Node are more strict than before. Could this not cause a similar issue?

Reply:
Not that I have noticed.